File: CS/ContextNavigation.Module/BusinessClasses.cs
using System;
using System.ComponentModel;

using DevExpress.Xpo;
using DevExpress.Xpo.Metadata;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;


namespace ContextNavigation.Module {
    public class TypeValueConverter : ValueConverter {
        public override object ConvertToStorageType(object value) {
            return value.ToString();
        }
        public override object ConvertFromStorageType(object value) {
            if(value == null) return null;
            else return Type.GetType((string)value);
        }
        public override Type StorageType {
            get { return typeof(string); }
        }
    }

    [DefaultClassOptions, ImageName("BO_Report")]
    public class HelpDocument : BaseObject {
        public HelpDocument(Session session) : base(session) { }

        [DevExpress.Xpo.ValueConverter(typeof(TypeValueConverter))]
        public Type ObjectType {
            get { return GetPropertyValue<Type>("ObjectType"); }
            set { SetPropertyValue<Type>("ObjectType", value); }
        }
        public string Title {
            get { return GetPropertyValue<string>("Title"); }
            set { SetPropertyValue<string>("Title", value); }
        }
        [Size(-1)]
        public string Text {
            get { return GetPropertyValue<string>("Text"); }
            set { SetPropertyValue<string>("Text", value); }
        }
    }

    [DefaultClassOptions]
    public class Contact : BaseObject {
        public Contact(Session session) : base(session) { }

        public string Name {
            get { return GetPropertyValue<string>("Name"); }
            set { SetPropertyValue<string>("Name", value); }
        }
    }

    [DefaultClassOptions]
    public class Task : BaseObject {
        public Task(Session session) : base(session) { }

        public string Description {
            get { return GetPropertyValue<string>("Description"); }
            set { SetPropertyValue<string>("Description", value); }
        }
    }
}

File: VB/ContextNavigation.Module/BusinessClasses.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel

Imports DevExpress.Xpo
Imports DevExpress.Xpo.Metadata

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation


Namespace ContextNavigation.Module
    Public Class TypeValueConverter
        Inherits ValueConverter
        Public Overrides Function ConvertToStorageType(ByVal value As Object) As Object
            Return value.ToString()
        End Function
        Public Overrides Function ConvertFromStorageType(ByVal value As Object) As Object
            If value Is Nothing Then
                Return Nothing
            Else
                Return Type.GetType(CStr(value))
            End If
        End Function
        Public Overrides ReadOnly Property StorageType() As Type
            Get
                Return GetType(String)
            End Get
        End Property
    End Class

    <DefaultClassOptions, ImageName("BO_Report")> _
    Public Class HelpDocument
        Inherits BaseObject
        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        <DevExpress.Xpo.ValueConverter(GetType(TypeValueConverter))> _
        Public Property ObjectType() As Type
            Get
                Return GetPropertyValue(Of Type)("ObjectType")
            End Get
            Set(ByVal value As Type)
                SetPropertyValue(Of Type)("ObjectType", value)
            End Set
        End Property
        Public Property Title() As String
            Get
                Return GetPropertyValue(Of String)("Title")
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Title", value)
            End Set
        End Property
        <Size(-1)> _
        Public Property Text() As String
            Get
                Return GetPropertyValue(Of String)("Text")
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Text", value)
            End Set
        End Property
    End Class

    <DefaultClassOptions> _
    Public Class Contact
        Inherits BaseObject
        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public Property Name() As String
            Get
                Return GetPropertyValue(Of String)("Name")
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Name", value)
            End Set
        End Property
    End Class

    <DefaultClassOptions> _
    Public Class Task
        Inherits BaseObject
        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public Property Description() As String
            Get
                Return GetPropertyValue(Of String)("Description")
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Description", value)
            End Set
        End Property
    End Class
End Namespace
